language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1451,83 +1455,118 @@ private static String getQualifiedModuleName(Module module) {
-        return resolveParameterSymbol(functionTypeSymbol, ctx, node.arguments());
+    public static Optional<ParameterSymbol> resolveParameterSymbol(FunctionTypeSymbol functionTypeSymbol,
+                                                                   PositionedOperationContext ctx,
+                                                                   NodeList<FunctionArgumentNode> arguments) {
+        return resolveParameterSymbol(functionTypeSymbol, ctx, arguments, false);
     }
 
-    /**
-     * Given the cursor position information, returns the expected ParameterSymbol
-     * information corresponding to the FunctionTypeSymbol instance.
-     *
-     * @param functionTypeSymbol Referenced FunctionTypeSymbol
-     * @param ctx                Positioned operation context information.
-     * @param node               Remote method call action node.
-     * @return {@link Optional<ParameterSymbol>} Expected Parameter Symbol.
-     */
-    public static Optional<ParameterSymbol> resolveFunctionParameterSymbol(FunctionTypeSymbol functionTypeSymbol,
-                                                                           PositionedOperationContext ctx,
-                                                                           RemoteMethodCallActionNode node) {
-        return resolveParameterSymbol(functionTypeSymbol, ctx, node.arguments());
+
+    private static Optional<ParameterSymbol> resolveParameterSymbol(FunctionTypeSymbol functionTypeSymbol,
+                                                                   PositionedOperationContext ctx,
+                                                                   NodeList<FunctionArgumentNode> arguments,
+                                                                   boolean isLangLibFunction) {
+        int cursorPosition = ctx.getCursorPositionInTree();
+        int argIndex = 0;
+        for (Node child : arguments) {
+            if (child.textRange().endOffset() < cursorPosition) {
+                argIndex += 1;
+            }
+        }
+
+        Optional<List<ParameterSymbol>> parameterSymbols = functionTypeSymbol.params();
+        Optional<ParameterSymbol> restParam = functionTypeSymbol.restParam();
+
+        // Check if we are not in an erroneous state
+        if (parameterSymbols.isEmpty() || parameterSymbols.get().size() < argIndex + 1) {
+            return Optional.empty();
+        }
+
+        // If the function is a lang lib method, need to add 1 to skip the 1st parameter which is the same type.
+        if (isLangLibFunction) {
+            argIndex = argIndex + 1;
+        }
+
+        // We can be in required params or rest params
+        if (parameterSymbols.get().size() > argIndex) {
+            return Optional.of(parameterSymbols.get().get(argIndex));
+        } else if (restParam.isPresent()) {
+            return Optional.of(restParam.get());
+        }
```

Comment:
We can use `if` clause, instead of `else if`

Reply:
+1